src/common/meta/src/key.rs

```diff
@@ -62,6 +62,14 @@
 //! 13. Topic name to region map key `__topic_region/{topic_name}/{region_id}`
 //!     - Mapping {topic_name} to {region_id}
 //!
+//! 14. Consistency guard key `__consistency_guard/{resource_type}/{resource_id}`
```

Comment:
Why do we need another lock when we already have the table lock?

Reply:
The key is used to indicate an ongoing operation on a specific resource, ensuring metadata consistency between datanodes and the metadata. If any potential operation failure (e.g., a partial datanode returns success) could lead to metadata inconsistency, any attempt to alter the table will be rejected until the metadata is consistent again, which may require manual table repair.

Reply:
My question is why we can just use the existing table metadata key?

```
//! 5. Table name key: `__table_name/{catalog_name}/{schema_name}/{table_name}`
//!     - The value is a [TableNameValue] struct; it contains the table id.
//!     - Used in the table name to table id lookup.
```

Reply:
I prefer not to mix them together.

Reply:
I don't think it's a mix. Remember that we designed the table metadata key to protect the table's metadata and ensure its consistency.  And this is a consistent issue, am I right?